### ISSUE TITLE
Applciation crash

### ISSUE THREAD

**Issue by zenker** [ChimeraTK/DeviceAccess-DoocsBackend#2]
With the latest version 00.12. my application crashes when shutting down.

BT:
*** Error in `./llrfserver': double free or corruption (!prev): 0x00007f66ac00cc00 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x777e5)[0x7f66cb13e7e5]
/lib/x86_64-linux-gnu/libc.so.6(+0x8037a)[0x7f66cb14737a]
/lib/x86_64-linux-gnu/libc.so.6(cfree+0x4c)[0x7f66cb14b53c]
/usr/lib/libChimeraTK-DeviceAccess-DoocsBackend.so.llrfserver(_ZN6cppext12future_queueI6EqDataNS_9MOVE_DATAEE14push_overwriteIS1_Li0EEEbOT_+0x93)[0x7f66c6bcf6c3]
/usr/lib/libChimeraTK-DeviceAccess-DoocsBackend.so.llrfserver(_ZN6cppext12future_queueI6EqDataNS_9MOVE_DATAEE14push_overwriteIS1_Li0EEEbRKT_+0x2f)[0x7f66c6bcf99f]
/export/doocs/lib/libEqServer.so.18.11.5-xenial1(dmsg_rcv+0xae5)[0x7f66c683e055]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x76ba)[0x7f66cd3b96ba]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x6d)[0x7f66cb1ce41d]

**Comment by mhier** [comment]
Are you sure the device access version used to compile the application and the backend are identical? Those crashes might happen if they are different. We are still looking for a clean solution to this problem, but it is more a device access issue...

**Comment by zenker** [comment]
I'm not totally sure, but both - my application and the doocsbackend - packages require libchimeratk-deviceaccess01-00-xenial1. So I guess both are compiled using the same version of DeviceAccess.

**Comment by killenb** [comment]
It is not clear where this is coming from. We did the following tests:
- QtHardMon with DoocsBackend, reading MACRO_PULSE_NUMBER without "wait for new data"
-> Works, no crash
- Simple C++ code reading MACRO_PULSE_NUMBER with "wait for new data" flag
-> Works, no crash
- Super simple ApplicationCore program just publishing MACRO_PULSE_NUMBER to the CS.
  I moved the TimingDummy backend from the example to an so-file for comparison
    - linked with OPC UA adapter
    -> runs fine but crashes on exit
    - linked with Doocs adapter without config file
    -> crashes on exit, after reporting exit due to missing config file
    - linked with Doocs adapter with config file
    -> reports 'safe exit' but never stops the process. has to be killed with SIGKILL
   - linked with Doocs adapter, loading TimingDummy and DoocsBackend from dmap file, using dummy
   -> works, no crash

So only in application core if the backend is used there are problems, with different results with DOOCS (hangs) and OPC UA (crashes) when exiting.
